dod-advana/gamechanger-web
# Task/homepage improvement testing

## Description
Updated New Homepage
- speed updates using redis caching for img data
- default images implemented to avoid dead image links
- Popular publications implemented
- admin control page for popular publications implemented
- bugfix: clicking on publications now links to document details page correctly
- bugfix: formatting search Time string in recent searches
- bugfix: long titles have been truncated to 50 characters, at the end of the nearest word. also using Title Case for text.
## Related Issue/Ticket

https://jira.di2e.net/browse/UOT-109748
https://jira.di2e.net/browse/UOT-107944

## Testing instructions

- go to http://10.194.9.112:8080/#/gamechanger-test
- check load times are bearable to the average human
- check loading behavior, default images are used until the actual images load + swap in. 
- click on a source. verify it links to a source details page correctly
- hover over any publication. check to see if the Title Case is used, and if long titles are truncated well. 
- click on a publication. verify that it goes to the details page correctly 
- go to admin -> homepage editor. there you should see a table to enable / disable entries in popular publications. check that both disabling entries + re-enabling entries works as expected (make sure to to hit save after making changes to that table)

## Unit Testing
backend/test/controllers/documentController.test.js
#getHomepageThumbnail
backend/test/utils/searchUtility.test.js
#getPopularQuery
